datalad/distribution/tests/test_update.py

```diff
@@ -89,6 +94,18 @@ def test_update_simple(origin, src_path, dst_path):
     ok_file_has_content(opj(dest.path, 'subm 2', 'load.dat'), 'heavy')
 
 
+@with_tempfile
+@with_tempfile
+def test_update_git_smoke(src_path, dst_path):
+    # Apparently was just failing on git repos for basic lack of coverage, hence this quick test
+    ds = Dataset(src_path).create(no_annex=True)
+    target = install(dst_path, source=src_path)
+    create_tree(ds.path, {'file.dat': '123'})
+    ds.add('file.dat')
+    target.update(recursive=True, merge=True)
```

Comment:
why not? ;-)  grew organically while me trying to hit that failing logic nail, and I was ready to add a sub-dataset

I just know that in this setting it was failing before, and passes now